neptunecontrib/monitoring/xgboost_monitor.py

```diff
@@ -0,0 +1,219 @@
+#
+# Copyright (c) 2020, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import tempfile
+
+import neptune
+import xgboost as xgb
+
+
+def neptune_callback(log_model=True,
+                     log_importance=True,
+                     max_num_features=None,
+                     log_tree=(0,),
+                     **kwargs):
+    """XGBoost callback for Neptune experiments.
+
+    This is XGBoost callback that automatically logs training and evaluation metrics, feature importance chart,
+    visualized trees and trained Booster to Neptune.
+
+    Make sure you created an experiment before you start XGBoost training using ``neptune.create_experiment()``
+    (`check our docs <https://docs.neptune.ai/neptune-client/docs/project.html
+    #neptune.projects.Project.create_experiment>`_).
+
+    Integration works with ``xgboost>=0.82``.
+
+    Note:
+        Check Neptune documentation for the `full example <https://docs.neptune.ai/integrations/xgboost.html>`_.
```

Comment:
Maybe we could add a direct link to some experiment and full code in Neptune just in case somebody doesn't want to go to the docs first.

Reply:
I put link to google colab, so user can go straight to the demo.